zanzibar/app.py

```diff
@@ -15,88 +21,103 @@ def get_books():
 @app.route('/namespace', methods=['POST'])
 @require_api_key
 def add_namespace():
-    data = request.json
+    data = sanitize_input(request.json)
     try:
         namespace_service.add(data)
+        app_logger.info(f"Namespace config added: {data}")
         return jsonify({}), 200
     except:
+        security_logger.warning(f"Bad request from IP: {request.remote_addr}")
         return jsonify({'Bad request!'}), 400
 
 @app.route('/namespace', methods=['GET'])
 @require_api_key
 def get_config():
-    namespace = request.args.get('namespace')
+    namespace = sanitize_input(request.args.get('namespace'))
```

Comment:
Consider adding encryption before logging.

Reply:
Sure, added! Check it out now :)

Reply:
Nice! I will merge now.